ui/src/components/tag/AddTagForm.js

```diff
@@ -0,0 +1,223 @@
+
+export default class AddTagForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = props.api;
+        this.addTagValue = this.addTagValue.bind(this);
+
+        if (props.editedTagKey && props.editedTagValues) {
+            this.state = {
+                newTagValue: '',
+                tagName: props.editedTagKey,
+                tagValues: props.editedTagValues,
+            };
+        } else {
+            this.state = {
+                newTagValue: '',
+                tagName: '',
+                tagValues: [],
+            };
+        }
+    }
+
+    inputChanged(key, evt) {
+        let value = '';
+        if (evt.target) {
+            value = evt.target.value;
+        } else {
+            value = evt ? evt : '';
+        }
+        this.setState({ [key]: value }, () =>
+            this.props.onUpdate(this.state.tagName, this.state.tagValues)
+        );
+    }
+
+    addTagValue() {
+        let name = this.state.newTagValue;
+        let tagValues = this.state.tagValues;
+
+        if (!name) {
+            return;
+        }
+        let names = (name || '')
+            .replace(/[\r\n\s]+/g, ',')
+            .split(',')
+            .map((n) => n.trim())
+            .filter((n) => n);
+
+        for (let i = 0; i < names.length; i++) {
+            if (tagValues.indexOf(names[i]) === -1) {
+                tagValues.push(names[i]);
+            }
+        }
+
+        this.setState(
+            {
+                tagValues,
+                newTagValue: '',
+            },
+            () => this.props.onUpdate(this.state.tagName, this.state.tagValues)
+        );
+    }
+
+    removeValue(idx) {
+        let tagValues = this.state.tagValues;
+        tagValues.splice(idx, 1);
+        this.setState({ tagValues }, () =>
+            this.props.onUpdate(this.state.tagName, this.state.tagValues)
+        );
+    }
+
+    render() {
+        let tagValueChanged = this.inputChanged.bind(this, 'newTagValue');
+        let tagNameChanged = this.inputChanged.bind(this, 'tagName');
+
+        let tagValues = this.state.tagValues
+            ? this.state.tagValues.map((val, idx) => {
+                  let remove = this.removeValue.bind(this, idx);
+                  return (
+                      <StyledTagColor key={val} onClickRemove={remove}>
+                          <StyledAnchor style={StyledAnchorActiveInline}>
+                              {' '}
+                              {val}{' '}
+                          </StyledAnchor>
+                      </StyledTagColor>
+                  );
+              })
+            : '';
+
+        return (
+            <SectionsDiv autoComplete={'off'} data-testid='add-tag-form'>
+                <SectionDiv>
+                    <StyledInputLabel htmlFor='tag-name'>
+                        Tag Name
+                    </StyledInputLabel>
+                    <ContentDiv>
+                        <StyledInput
```

Comment:
if the current operation is to edit the tag, let us not show input for the tag name, instead show a text field with the tag name which won't be editable or disable this input field